src/rebar3_appup_generate.erl

```diff
@@ -206,15 +206,15 @@ generate_instruction(deleted, File) ->
 
 generate_instruction(changed, ModDeps, {File, _}) ->
     {ok, {Name, List}} = beam_lib:chunks(File, [attributes, exports]),
-    Behavior = get_behavior(List),
+    Supervisor = is_supervisor(List),
     CodeChange = is_code_change(List),
     Deps = proplists:get_value(Name, ModDeps, []),
-    generate_instruction_advanced(Name, Behavior, CodeChange, Deps).
+    generate_instruction_advanced(Name, Supervisor, CodeChange, Deps).
 
 generate_instruction_advanced(Name, undefined, undefined, Deps) ->
     %% Not a behavior or code change, assume purely functional
     {load_module, Name, Deps};
-generate_instruction_advanced(Name, [supervisor], _, _) ->
+generate_instruction_advanced(Name, supervisor, _, _) ->
     %% Supervisor
```

Comment:
`generate_instruction_advanced` could take a list of behaviours and generate the corresponding appup instruction, that way we'll be able to support multiple behaviours in a module